gnome-initial-setup/setlocale-wrapper.c

```diff
@@ -0,0 +1,52 @@
+/* -*- mode: C; c-file-style: "gnu"; indent-tabs-mode: nil; -*- */
+/*
+ * Copyright (C) 2017 Endless Mobile, Inc.
+ *
+ * Authors:
+ *   Joaquim Rocha <jrocha@endlessm.com>
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation; either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, see <http://www.gnu.org/licenses/>.
+ */
+
+#define _GNU_SOURCE
+
+#include <string.h>
+#include <dlfcn.h>
+
+#define WRAPPER_LOCALE_BUF_LEN 16
```

Comment:
I didn't think so. Because it's the LCID code + _UTF-8_ AFAICS. e.g. _pt_PT.UTF-8_ which is a maximum of 11 chars but I rounded up to 16 as it's a better number as 
https://en.wikipedia.org/wiki/Locale_(computer_software)

However I have just remembered the ca_ES.UTF-8**@valencia** special locale... I guess we can round it up to 32 bytes just to be sure?